Remove Idemixgen from Fabric produced binaries

Sayalikukkar opened the pull request:
<!--- DELETE MARKDOWN COMMENTS BEFORE SUBMITTING PULL REQUEST. -->

<!--- Provide a descriptive summary of your changes in the Title above. -->

#### Type of change

<!--- What type of change? Pick one option and delete the others. -->

- Documentation update

#### Description
Removed Idemixgen from the documentation to reflect its removal from the set of Fabric produced binaries. Also created the seperate PR for idemix repo.

<!--- Describe your changes in detail, including motivation. -->

#### Additional details

<!--- Additional implementation details or comments to reviewers. -->
<!--- Summarize how the pull request was tested (if not obvious from commit). -->

#### Related issues

Closes https://github.com/hyperledger/fabric/issues/3575

<!---
#### Release Note
If change impacts current users, uncomment Release Note heading and provide
release note text.
Also, copy release note text into the release specific /release_notes file.
-->

<!--
Checklist (DELETE AFTER READING):

- `Signed-off-by` added to commits (required for DCO check to pass)
- Tests have been added/updated (required for bug fixes and features)
- Unit and/or integration tests pass locally
- Run linters and checks locally using 'make checks'
- If change requires documentation updates, make updates in pull request,
  or open a separate issue and provide link
- Squash commits into a single commit, unless a stack of commits is
  intentional to assist reviewers or to preserve review comments.
- For additional contribution guidelines see the project's CONTRIBUTING.md file
-->

Sayalikukkar committed:
Remove Idemixgen from Fabric produced binaries

Signed-off-by: sayalikukkar <sayukukkar11@gmail.com>

denyeart commented:
Thank you for the update.

There is also a link to this document in a table of contents that needs to be removed at the same time - https://raw.githubusercontent.com/hyperledger/fabric/main/docs/source/ops_guide.rst

Also https://github.com/hyperledger/fabric/blob/main/docs/source/idemix.rst mentions that Fabric provides this tool. I think we'd want to update that wording from "Fabric provides two possible Idemix issuers" to "There are two possible Idemix issuers". And make "idemix tool" a hyperlink over to https://github.com/IBM/idemix.

Finally, I think we'd want this doc content to appear at https://github.com/IBM/idemix where the tool now resides, before removing it from Fabric docs, do you want to submit the PR to https://github.com/IBM/idemix as well to complete the move?

Sayalikukkar commented:
@denyeart , Sure, I'm interested in helping with that. I can make the necessary updates to the Fabric documentation as you've described and also submit a pull request to the IBM idemix repository to ensure the content is properly moved.

Sayalikukkar committed:
Remove Idemixgen from Fabric produced binaries

Signed-off-by: sayalikukkar <sayukukkar11@gmail.com>

Remove Idemixgen from Fabric produced binaries

Signed-off-by: sayalikukkar <sayukukkar11@gmail.com>

Remove Idemixgen from Fabric produced binaries

Signed-off-by: sayalikukkar <sayukukkar11@gmail.com>

denyeart commented on docs/source/ops_guide.rst:
```diff
@@ -1,27 +0,0 @@
-   pluggable_endorsement_and_validation
-   access_control.md
-   idemix
-   idemixgen
```
I meant to simply remove the idemixgen entry in this file, not the entire file!

denyeart commented:
Thank you, looks good to me now.
But let's wait for @ale-linux to merge https://github.com/IBM/idemix/pull/41 to ensure we have the doc somewhere.

Sayalikukkar commented:
Thank you for your feedback and guidance!

denyeart commented on docs/source/idemix.rst:
```diff
@@ -56,10 +56,10 @@ Fabric components correspond to the user, issuer, and verifier in Idemix.
 
    a) Fabric CA for production environments or development, and
-   b) the :doc:`idemixgen <idemixgen>` tool for development environments.
+   b) the `idemix <https://github.com/IBM/idemix>`_ tool for development environments.
```
```suggestion
   b) the `idemixgen <https://github.com/IBM/idemix/blob/main/tools/idemixgen/idemixgen.rst>`_ tool for development environments.
```

Ok, the idemixgen tool doc was merged into the idemix repository, so let's point directly to that now.

denyeart reviewed (approved):
Thank you for the quick responses, looks good now!

denyeart commented:
Sorry, there is now a conflict in the doc, I think in the ops_guide table of contents. Could you resolve the conflict by rebasing locally and then re-push?

Also, it would be nice to squash the 3 commits into a single commit, see guidance at https://hyperledger-fabric.readthedocs.io/en/latest/github/github.html#updating-a-pull-request.

denyeart commented:
Perfect, thank you!
